src/edu/washington/escience/myria/api/encoding/ParallelDatasetEncoding.java

```diff
@@ -0,0 +1,24 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.api.encoding;
+
+import java.util.Set;
+
+import edu.washington.escience.myria.RelationKey;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.io.AmazonS3Source;
+
+/**
+ *
+ */
+public class ParallelDatasetEncoding extends MyriaApiEncoding {
+  @Required public RelationKey relationKey;
+  @Required public Schema schema;
+  @Required public AmazonS3Source s3Source;
+  public Character delimiter;
+  public Character escape;
+  public Integer numberOfSkippedLines;
+  public Character quote;
+  public Set<Integer> workers;
```

Comment:
Yes, that was an idiotic suggestion. Apparently I'd forgotten what "range partitioning" means. Maybe we should recycle RoundRobinPartitionFunction since it shouldn't make any difference to the optimizer.